ark/jsonschema/composition.ts

```diff
@@ -0,0 +1,60 @@
+import { printable } from "@ark/util"
+import { type, type JsonSchema, type Type } from "arktype"
+import { parseJsonSchema } from "./json.ts"
+
+const parseAllOfJsonSchema = (jsonSchemas: readonly JsonSchema[]): Type =>
+	jsonSchemas
+		.map(jsonSchema => parseJsonSchema(jsonSchema))
+		.reduce((acc, validator) => acc.and(validator))
+
+const parseAnyOfJsonSchema = (jsonSchemas: readonly JsonSchema[]): Type =>
+	jsonSchemas
+		.map(jsonSchema => parseJsonSchema(jsonSchema))
+		.reduce((acc, validator) => acc.or(validator))
+
+const parseNotJsonSchema = (jsonSchema: JsonSchema) => {
+	const inner = parseJsonSchema(jsonSchema)
+	return type.unknown.narrow((data, ctx) =>
+		inner.allows(data) ?
+			ctx.reject({
+				expected: `a value that's not ${inner.description}`,
```

Comment:
maybe we shorten this to just `not inner.description`?

if there are cases where that sounds really bad the current version is fine, but I'd be okay with `must be not a string` or similar in most cases

Reply:
Done in a16e4771